atc/exec/set_pipeline_step.go

```diff
@@ -0,0 +1,229 @@
+package exec
+
+import (
+	"context"
+	"fmt"
+	"github.com/concourse/baggageclaim"
+	"io/ioutil"
+	"os"
+	"sigs.k8s.io/yaml"
+	"strings"
+
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagerctx"
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/atc/creds"
+	"github.com/concourse/concourse/atc/db"
+	"github.com/concourse/concourse/atc/db/lock"
+	"github.com/concourse/concourse/atc/exec/artifact"
+	"github.com/concourse/concourse/vars"
+)
+
+//go:generate counterfeiter . SetPipelineDelegate
+
+type SetPipelineDelegate interface {
+	BuildStepDelegate
+
+	Initializing(lager.Logger)
+	Starting(lager.Logger)
+	Finished(lager.Logger, ExitStatus, db.ConfigVersion, db.ConfigVersion)
+}
+
+// TaskStep executes a TaskConfig, whose inputs will be fetched from the
+// artifact.Repository and outputs will be added to the artifact.Repository.
+type SetPipelineStep struct {
+	planID      atc.PlanID
+	plan        atc.SetPipelinePlan
+	metadata    StepMetadata
+	delegate    SetPipelineDelegate
+	dbConn      db.Conn
+	lockFactory lock.LockFactory
+	succeeded   bool
+}
+
+func NewSetPipelineStep(
+	planID atc.PlanID,
+	plan atc.SetPipelinePlan,
+	metadata StepMetadata,
+	delegate SetPipelineDelegate,
+	dbConn db.Conn,
+	lockFactory lock.LockFactory,
+) Step {
+	return &SetPipelineStep{
+		planID:   planID,
+		plan:     plan,
+		metadata: metadata,
+		//workerClient: workerClient,
+		delegate:    delegate,
+		dbConn:      dbConn,
+		lockFactory: lockFactory,
+	}
+}
+
+func (step *SetPipelineStep) Run(ctx context.Context, state RunState) error {
+	logger := lagerctx.FromContext(ctx)
+	logger = logger.Session("set-pipeline-step", lager.Data{
+		"step-name": step.plan.Name,
+		"job-id":    step.metadata.JobID,
+	})
+
+	step.delegate.Initializing(logger)
+
+	variables := step.delegate.Variables()
+	params, err := creds.NewParams(variables, step.plan.Params).Evaluate()
+	if err != nil {
+		return err
+	}
+
+	spParams, err := atc.NewSetPipelineParams(params)
+	if err != nil {
+		return err
+	}
+	repository := state.Artifacts()
+
+	artifactSource := setPipelineArtifactSource{
+		ctx:    ctx,
+		logger: logger,
+		params: spParams,
+		repo:   repository,
+	}
+
+	atcConfig, err := artifactSource.FetchConfig()
+	if err != nil {
+		return err
+	}
+
+	step.delegate.Starting(logger)
+
+	if spParams.PipelineName == "self" {
+		spParams.PipelineName = step.metadata.PipelineName
+	}
+
+	team, found, err := db.NewTeamFactory(step.dbConn, step.lockFactory).FindTeam(step.metadata.TeamName)
+	if err != nil {
+		logger.Error("EVAN-failed-to-find-team", err)
```

Comment:
I guess this and the following debug messages starting with `EVAN-` are not meant to be final? :)

Reply:
Of course not. This is still under WIP. :)

Reply:
Sorry, my bad, I've missed that :)